src/els_parser.erl

```diff
@@ -397,11 +394,29 @@ record_expr(Tree) ->
   case erl_syntax:type(RecordNode) of
     atom ->
       Record = erl_syntax:atom_value(RecordNode),
-      [poi(erl_syntax:get_pos(Tree), record_expr, Record)];
+      [ poi(erl_syntax:get_pos(Tree), record_expr, Record)
+      | record_expr_fields(Record, Tree)];
     _ ->
       []
   end.
 
+-spec record_expr_fields(atom(), tree()) -> [poi()].
+record_expr_fields(Record, RecordExprTree) ->
+  Fields = erl_syntax:record_expr_fields(RecordExprTree),
+  [begin
+     NameNode = erl_syntax:record_field_name(FieldNode),
+     NameAtom = record_field_name_atom(NameNode),
+     poi(erl_syntax:get_pos(NameNode), record_field, NameAtom, Record)
```

Comment:
To answer your question in the description, an alternative could be to have a composite `Id`, something like `{RecordName, FieldName}`, mostly for consistency with the other cases. The `Data` field is currently only used for storing the `pretty_printed` version of a spec.